### DISCUSSION TITLE
Can I control a group of `ui.upload` widgets with one single button?

### DISCUSSION THREAD

**Discussion by logandoo** [zauberzeug/nicegui#3345] [Q&A] [answered]
### Question

Is there any possibility that I can control a group of `ui.upload` widgets with one single button? Like, I have three different upload widgets in a single page, and I want to put a "submit" button in the page to submit all the uploaded files, rather than click the cloud button on each upload widget, or set them to auto_upload mode, since I  probably need to edit the file queue before I decide to upload them.

**Comment by python-and-novella** [comment, marked as answer]
both auto_upload and edit queue before upload them ?I'm confused about this.
you can set `auto-upload` `multiple` and `hide-upload-btn`, define `on_upload` of each ui.upload.

Here is other button controls upload.
```python3
from nicegui import events, ui

with ui.dialog().props('full-width') as dialog:
    with ui.card():
        content = ui.markdown()

def handle_upload(e: events.UploadEventArguments):
    text = e.content.read().decode('utf-8')
    content.set_content(text)
    dialog.open()

with ui.element():
    up = ui.upload(on_upload=handle_upload).props('').classes('max-w-full')

ui.button('upload',on_click=lambda :up.run_method('upload'))

ui.run()
```

**Reply by logandoo** [reply]
Sorry for the confusion, just dismiss the text after the first sentence. Your answer is perfect for me.
